### PR TITLE
feat(order): add Magento guest orders driver

### PR DESCRIPTION
**Pull request by griest024**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Fixes: N/A

## What is the new behavior?
- Adds error map and transform
- Adds validators for guest orders response
- Adds order transformer
- Adds guest order driver service

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: libs/order/src/drivers/src/magento/errors/codes.ts [hunk 1]
```diff
@@ -0,0 +1,3 @@
+export enum MagentoOrderGraphQlErrorCode {
+	CART_NOT_FOUND = 'graphql-no-such-entity',
```
Comment (lderrickable) [inline libs/order/src/drivers/src/magento/errors/codes.ts:2]: Could we reuse the error code `@daffodil/cart`? It's the same code

File: libs/order/src/drivers/src/magento/order.service.spec.ts [hunk 2]
```diff
@@ -0,0 +1,455 @@
...
+
+          op.flush({
+            data: mockGetOrdersResponse
+          });
+        });
+      });
+    });
+
+    describe('when the call to the Magento API is unsuccessful', () => {
+      it('should throw a DaffCartNotFoundError', done => {
```
Comment (lderrickable) [inline libs/order/src/drivers/src/magento/order.service.spec.ts:428]: I don't think you need to test the error transform logic here, but I think this "it" block should just be "should throw an error" or something more general. Just to be clear that this won't necessarily throw a DaffCartNotFoundError.

File: libs/order/src/drivers/src/magento/errors/codes.ts (hunk 1 above)
Comment (griest024) [reply libs/order/src/drivers/src/magento/errors/codes.ts:2]: I don't think it gets exported.

File: libs/order/src/drivers/src/magento/order.service.spec.ts (hunk 2 above)
Comment (griest024) [reply libs/order/src/drivers/src/magento/order.service.spec.ts]: changed

**Pull request merged**